src/core/thread/mle.cpp

```diff
@@ -712,16 +712,6 @@ void Mle::HandleUnicastAddressesChanged(void)
 
         mNetif.AddUnicastAddress(mMeshLocal64);
     }
-
-    switch (mDeviceState)
-    {
-    case kDeviceStateChild:
-        SendChildUpdateRequest();
```

Comment:
I agree that we should not send a Child Update Request just because an RLOC changed.  However, it is important to send a Child Update Request whenever a non-RLOC IPv6 address is configured.